agent-manager-service/services/infra_resource_manager.go

```diff
@@ -226,282 +270,231 @@ func (s *infraResourceManager) ListProjects(ctx context.Context, userIdpId uuid.
 		projectResponses = append(projectResponses, projectResponse)
 	}
 
-	s.logger.Info("Fetched projects successfully", "orgName", orgName)
+	s.logger.Info("Fetched projects successfully", "orgName", orgName, "count", len(projectResponses), "total", total)
 	return projectResponses, int32(total), nil
 }
 
 func (s *infraResourceManager) DeleteProject(ctx context.Context, userIdpId uuid.UUID, orgName string, projectName string) error {
+	s.logger.Debug("DeleteProject called", "userIdpId", userIdpId, "orgName", orgName, "projectName", projectName)
+
 	// Validate organization exists
 	org, err := s.OrganizationRepository.GetOrganizationByOrgName(ctx, userIdpId, orgName)
 	if err != nil {
 		if db.IsRecordNotFoundError(err) {
+			s.logger.Debug("Organization not found", "userIdpId", userIdpId, "orgName", orgName)
 			return utils.ErrOrganizationNotFound
 		}
+		s.logger.Error("Failed to get organization from repository", "userIdpId", userIdpId, "orgName", orgName, "error", err)
 		return fmt.Errorf("failed to find organization %s: %w", orgName, err)
 	}
 
 	project, err := s.ProjectRepository.GetProjectByName(ctx, org.ID, projectName)
 	if err != nil {
 		// DELETE is idempotent
 		if db.IsRecordNotFoundError(err) {
+			s.logger.Debug("Project not found, treating as successful delete (idempotent)", "orgName", orgName, "projectName", projectName)
 			return nil
 		}
+		s.logger.Error("Failed to get project from repository", "orgId", org.ID, "projectName", projectName, "error", err)
 		return fmt.Errorf("failed to find project %s: %w", projectName, err)
 	}
+	s.logger.Debug("Project found", "orgName", orgName, "projectName", projectName, "projectId", project.ID)
+
 	// Check agents exist for the project
+	s.logger.Debug("Checking for associated agents", "projectId", project.ID, "projectName", projectName)
 	agents, err := s.AgentRepository.ListAgents(ctx, org.ID, project.ID)
 	if err != nil {
+		s.logger.Error("Failed to list agents for project", "projectId", project.ID, "projectName", projectName, "error", err)
 		return fmt.Errorf("failed to list agents for project %s: %w", projectName, err)
 	}
 	if len(agents) > 0 {
+		s.logger.Warn("Cannot delete project with associated agents", "orgName", orgName, "projectName", projectName, "agentCount", len(agents))
 		return fmt.Errorf("cannot delete project %s: project has %d associated agent(s)", projectName, len(agents))
 	}
-	// Soft delete the project from the database
-	if err := s.ProjectRepository.SoftDeleteProject(ctx, org.ID, project.ID); err != nil {
-		return fmt.Errorf("failed to delete project %s from repository: %w", projectName, err)
-	}
+	s.logger.Debug("No associated agents found, proceeding with deletion", "projectName", projectName)
 
-	// Delete project from OpenChoreo
+	// Delete project from OpenChoreo first
 	if err := s.OpenChoreoSvcClient.DeleteProject(ctx, orgName, projectName); err != nil {
```

Comment:
Isn't it better to do a soft delete -> delete in openchoreo

if openchoreo delete is successfull do the harddelete
otherwise undo the soft delete